tests/kernel/irq_lock/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  kernel.test:
+    arch_exclude: x86 posix xtensa nios2 riscv32
```

Comment:
why are we excluding most architectures here?

Reply:
For arch x86 and posix:
During compilation of test case for architecture like x86 and posix, it was giving error:
error: CONFIG_NUM_IRQS undeclared (first use in this function)

CONFIG_NUM_IRQS config is not available in Kconfig for arch x86 and posix.
We are using this config only in test applications not in drivers.
To avoid the compilation error in shippable, I excluded few architectures (x86 and posix) for this application.

For xtensa and nios2:
AFAIK For few architecture like xtensa and nios2 we do not have software trigger interrupt support.
Instead of excluding the architecture using testcase.yaml, now I am taking care of it in main.c itself.

For riscv32:
Test case is failing for this architecture, looks like irq_lock() is not behaving as expected.
Will create a git hub issue once this test application get merged.  And will remove riscv32 from testcase.yaml as well.